packages/marshal/src/marshal-justin.js

```diff
@@ -358,12 +360,18 @@ const decodeToJustin = (encoding, shouldIndent = false) => {
         case 'slot': {
           let { index, iface } = rawTree;
           index = Number(Nat(index));
-          if (iface === undefined) {
-            return out.next(`slot(${index})`);
-          } else {
+          if (index < slots.length) {
+            const slot = quote(slots[index]);
```

Comment:
The other places in this file we call `quote` (aka `JSON.stringify`) we know that we're providing it data that can be faithfully JSON.stringified. From your example, I suspect you have in mind only cases where the quotes slot value is a string. Would it be appropriate to limit to those cases with an assert? If not, can you think of any cheap test that a JSON.stringify will be full fidelity?

Reply:
I implemented full fidelity (nested decode instead of JSON)!